src/redux/actions/samples/updateSample.js

```diff
@@ -1,28 +1,40 @@
-import _ from 'lodash';
 import moment from 'moment';
+import saveSamples from './saveSamples';
 
 import {
+  SAMPLES_RESTORE,
   SAMPLES_UPDATE,
 } from '../../actionTypes/samples';
+import pushNotificationMessage from '../notifications';
+import errorTypes from './errorTypes';
 
 const updateSample = (
   sampleUuid,
   sample,
 ) => async (dispatch, getState) => {
-  const currentSample = getState().samples[sample.uuid];
-
-  if (_.isEqual(currentSample, sample)) return null;
+  const currentSampleState = getState().samples;
 
   // eslint-disable-next-line no-param-reassign
   sample.lastModified = moment().toISOString();
 
-  dispatch({
-    type: SAMPLES_UPDATE,
-    payload: {
-      sampleUuid,
-      sample,
-    },
-  });
+  try {
+    dispatch({
+      type: SAMPLES_UPDATE,
+      payload: {
+        sampleUuid,
+        sample,
+      },
+    });
+
+    dispatch(saveSamples(currentSampleState[sampleUuid].projectUuid));
+  } catch (e) {
+    pushNotificationMessage('error', errorTypes.SAVE_SAMPLES);
+
+    dispatch({
+      type: SAMPLES_RESTORE,
+      state: currentSampleState,
+    });
```

Comment:
try to not use restore, but rather not update the redux store until the save is successful